File: php/calculadoraphp/app/Calculadora.php
<?php

namespace App;

use ArithmeticError;

class Calculadora
{
    private $resultado;

    public function __construct()
    {
        $this->resultado = 0.0;
    }

    public function getResultado()
    {
        return $this->resultado;
    }

    public function setResultado($resultado)
    {
        $this->resultado = $resultado;

        return $this;
    }

    public function somar($a, $b)
    {
        $this->resultado = $a + $b;
    }
    
    public function subtrair($a, $b)
    {
        $this->resultado = $a - $b;
    }

    public function multiplicar($a, $b)
    {
        $this->resultado = $a * $b;
    }

    public function dividir($a, $b)
    {
        if($b == 0){
            throw new ArithmeticError("Impossivel operacao de divisao por zero");
        }
        $this->resultado = $a / $b;
    }
}

File: java/CalculadoraExemplo/src/main/java/com/mycompany/app/Calculadora.java

package com.mycompany.app;


public class Calculadora {
    
    private double resultado;

    public Calculadora() {
        this.resultado = 0.0;
    }
    
    public double getResultado(){
        return this.resultado;
    }
    
    public void setResultado(double r){
        this.resultado = r;
    }
    
    public void somar(double a, double b){
        this.resultado = a + b;
    }
    
    public void subtrair(double a, double b){
        this.resultado = a - b;
    }
    
    public void multiplicar(double a, double b){
        this.resultado = a * b;
    }
    
    public void dividir(double a, double b) throws ArithmeticException{
        
        if(b == 0) throw new ArithmeticException("Impossivel realizar operacao de divisao por zero");
        this.resultado = a / b;
    }
}
